js/storage.js

```diff
@@ -1247,11 +1247,13 @@ Storage.importTeam = function (buffer, teams) {
 			if (line.substr(0, 14) === 'Hidden Power [') {
 				var hptype = line.substr(14, line.length - 15);
 				line = 'Hidden Power ' + hptype;
-				var type = Dex.types.get(hptype);
-				if (!curSet.ivs && type) {
-					curSet.ivs = {};
-					for (var stat in type.HPivs) {
-						curSet.ivs[stat] = type.HPivs[stat];
+				if (parseInt(format[3], 10) || 6 < 7) {
```

Comment:
This statement is always true because of `|| 6 < 7`. Also it would be wise to add a comment here/on the next line explaining the purpose of the block eg: `// update IVs to match hidden power if format is gen 6 or earlier`

Reply:
The value only resolves to 6 if it fails to parse an int. I will put the first part in parentheses to make that more clear. This is due to other code defaulting the generation to gen 6 and I wanted to keep that going. I don't foresee this causing much of any issue as any format that doesn't include a generation tag in the name will always suffer greatly when interacting with the client. I will go ahead and add the comment.

Reply:
Yep just tested it. Needed those paretheses to make it resolve properly. Need to remember to do that by default no matter what. Nice catch